src/CodeGen_Internal.cpp

```diff
@@ -440,6 +443,15 @@ Expr long_div(const Expr &num, const Expr &den, const uint64_t *max_abs) {
     return simplify(common_subexpression_elimination(q));
 }
 
+
+Expr long_mod_round_to_zero(const Expr &a, const Expr &b,
```

Comment:
I think maybe `unsigned_long_div_round_to_zero` should return a struct/pair of the div (q) and mod (num after the loop) result. That might result in better code when both are used?

Reply:
Done